lib/puppet/provider/ini_setting/splunk.rb

```diff
@@ -14,6 +14,10 @@ class << self
   def self.file_path
     raise Puppet::Error, 'file_path must be set with splunk_config type before provider can be used' if @file_path.nil?
     raise Puppet::Error, 'Child provider class does not support a file_name method' unless respond_to?(:file_name)
-    File.join(@file_path, file_name)
+    @file_path
```

Comment:
This breaks `instances` (as the parent provider `instances` method no longer uses the correct path).  In turn this breaks all settings related to purging unmanaged resources.